include/deal.II/lac/sparse_matrix.templates.h

```diff
@@ -1265,24 +1273,25 @@ namespace internal
     template <typename number,
               typename InVector,
               typename OutVector>
-    number residual_sqr_on_subrange (const size_type    begin_row,
-                                     const size_type    end_row,
-                                     const number      *values,
-                                     const std::size_t  *rowstart,
-                                     const size_type   *colnums,
-                                     const InVector    &u,
-                                     const InVector    &b,
-                                     OutVector         &dst)
+    typename InVector::value_type
```

Comment:
Good question. Will change this. 

It turns out that the function is only ever called with InVector=OutVector right now, so it doesn't matter. But it's the right thing to do to change it.